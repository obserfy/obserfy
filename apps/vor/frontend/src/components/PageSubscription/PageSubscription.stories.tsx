import { FC } from "react"
import PageSubscription from "./PageSubscription"

export default {
  title: "Core/PageSubscription",
  component: PageSubscription,
  parameters: {
    componentSubtitle: "Just a simple PageSubscription",
  },
}

export const Basic: FC = () => <PageSubscription />
