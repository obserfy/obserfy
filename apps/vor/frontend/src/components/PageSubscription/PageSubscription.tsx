import { t, Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Box, Button, Card, Flex, useColorMode } from "theme-ui"
import { borderTop } from "../../border"
import dayjs from "../../dayjs"
import { useGetSchool } from "../../hooks/api/schools/useGetSchool"
import { useGetUserProfile } from "../../hooks/api/useGetUserProfile"
import { ReactComponent as CheckmarkIcon } from "../../icons/checkmark-outline.svg"
import { ReactComponent as CancelIcon } from "../../icons/close.svg"
import { ReactComponent as CreditCardIcon } from "../../icons/credit-card.svg"
import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import {
  ADMIN_SUBSCRIPTION_SUCCESS_URL,
  ADMIN_URL,
  ADMIN_USERS_URL,
} from "../../routes"
import Icon from "../Icon/Icon"
import { Link, navigate } from "../Link/Link"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import Typography from "../Typography/Typography"

export const PageSubscription: FC = () => {
  const [loading, setLoading] = useState(false)
  const user = useGetUserProfile()
  const school = useGetSchool()
  const [colorMode] = useColorMode()
  const isSubscribed = school.data?.subscription

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.xsm" }}>
      <TopBar
        breadcrumbs={[
          breadCrumb(t`Admin`, ADMIN_URL),
          breadCrumb(t`Plans & Billing`),
        ]}
      />

      {school.data?.subscription && (
        <Card m={3} px={4} sx={{ borderRadius: 16 }}>
          <Flex pt={4} pb={3} sx={{ alignItems: "center" }}>
            <Typography.Body
              color="textPrimary"
              sx={{ fontWeight: "bold", lineHeight: 1 }}
            >
              <Trans>Standard</Trans>
            </Typography.Body>
            <Typography.Body ml="auto">
              <Trans>Current Plan</Trans>
            </Typography.Body>
          </Flex>
          <Box mt={3}>
            <Typography.Body
              sx={{ fontSize: 1, color: "textMediumEmphasis", lineHeight: 1 }}
            >
              <Trans>Next billing date</Trans>
            </Typography.Body>
            <Typography.Body>
              {dayjs(school.data?.subscription?.nextBillDate).format(
                "D MMMM YYYY"
              )}
            </Typography.Body>
          </Box>
          <Box mt={3} mb={3}>
            <Typography.Body
              sx={{ fontSize: 1, color: "textMediumEmphasis", lineHeight: 1 }}
            >
              <Trans>Price</Trans>
            </Typography.Body>
            <Typography.Body>
              ${4 * school.data?.users?.length} ({school.data?.users?.length}{" "}
              users) / month
            </Typography.Body>
          </Box>

          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
          <a href={school.data?.subscription.updateUrl}>
            <Flex py={3} sx={{ alignItems: "center" }}>
              <Icon
                as={CreditCardIcon}
                size={24}
                color="textPrimary"
                fill="transparent"
              />
              <Typography.Body ml={3}>
                <Trans>Payment details</Trans>
              </Typography.Body>
              <Icon as={NextIcon} ml="auto" />
            </Flex>
          </a>

          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
          <a href={school.data?.subscription.cancelUrl}>
            <Flex py={3} pb={4} sx={{ alignItems: "center" }}>
              <Icon as={CancelIcon} size={24} fill="danger" />
              <Typography.Body ml={3}>
                <Trans>Cancel Plan</Trans>
              </Typography.Body>
              <Icon as={NextIcon} ml="auto" />
            </Flex>
          </a>
        </Card>
      )}

      {school.isSuccess && !school.data?.subscription && (
        <Card m={3} px={4} sx={{ borderRadius: 16 }}>
          <Flex py={4} sx={{ alignItems: "center" }}>
            <Typography.Body color="textPrimary" sx={{ fontWeight: "bold" }}>
              <Trans>Free Trial</Trans>
            </Typography.Body>
            <Typography.Body ml="auto">
              <Trans>Current Plan</Trans>
            </Typography.Body>
          </Flex>
        </Card>
      )}

      {school.isLoading && (
        <Box px={3} pt={3}>
          <LoadingPlaceholder
            sx={{ width: "100%", height: "16rem", borderRadius: 16 }}
          />
        </Box>
      )}

      <Card m={3} p={4} mt={3} sx={{ borderRadius: 16 }}>
        <Typography.Body
          color="textPrimary"
          sx={{ fontWeight: "bold", lineHeight: 1 }}
          mb={2}
        >
          <Trans>Standard Plan</Trans>
        </Typography.Body>

        <Flex sx={{ alignItems: "baseline" }}>
          <Typography.H3 sx={{ fontWeight: "bold", lineHeight: 1 }}>
            $4
          </Typography.H3>
          <Typography.Body ml={2} color="textMediumEmphasis">
            <Trans>User / Month</Trans>
          </Typography.Body>
        </Flex>

        <Typography.Body my={3}>
          <Trans>Get full access to all features.</Trans>{" "}
          <a href="https://obserfy.com/pricing">
            <u>
              <Trans>Learn more.</Trans>
            </u>
          </a>
        </Typography.Body>

        <Box my={3}>
          <Feature text={t`Unlimited students`} />
          <Feature text={t`Record observations`} />
          <Feature text={t`Create lesson plans`} />
          <Feature text={t`Track curriculum progress`} />
          <Feature text={t`Parent dashboard`} />
          <Feature text={t`Image gallery`} />
          <Feature text={t`Video gallery`} />
        </Box>

        <Box mt={3} mb={3} sx={{ ...borderTop }} pt={3}>
          <Flex>
            <Typography.Body>You currently have</Typography.Body>
            <Link
              to={ADMIN_USERS_URL}
              sx={{ fontSize: 1, textDecoration: "underline", ml: "auto" }}
            >
              <Trans>Manage users</Trans>
            </Link>
          </Flex>
          <Typography.Body sx={{ opacity: school.data ? 1 : 0.5 }}>
            {school.data?.users.length ?? "... "} User(s)
          </Typography.Body>
        </Box>

        <Box>
          <Typography.Body sx={{ fontWeight: "bold" }}>Total</Typography.Body>
          <Typography.Body
            ml="auto"
            sx={{ fontWeight: "bold", opacity: school.data ? 1 : 0.5 }}
          >
            {school.data ? school.data?.users.length * 4 : "... "}$ / month
          </Typography.Body>
        </Box>

        <Button
          variant={isSubscribed ? "outline" : "primary"}
          mt={4}
          p={3}
          sx={{ width: "100%", fontWeight: "bold", borderRadius: 16 }}
          disabled={
            user.isLoading || school.isLoading || isSubscribed !== undefined
          }
          onClick={() => {
            setLoading(true)
            const script = document.createElement("script")
            script.type = "text/javascript"
            script.onload = () => {
              setLoading(false)
              Paddle.Setup({ vendor: 112134 })
              Paddle.Checkout.open({
                product: 590592,
                email: user.data?.email,
                passthrough: JSON.stringify({ schoolId: user.data?.school.id }),
                allowQuantity: false,
                displayModeTheme: colorMode === "dark" ? "dark" : "light",
                quantity: school.data?.users?.length ?? 1,
                message:
                  "Quantity and price will be adjusted later based on your school's user count.",
                successCallback: () => {
                  setTimeout(() => {
                    navigate(ADMIN_SUBSCRIPTION_SUCCESS_URL)
                  }, 500)
                },
              })
            }
            script.src = "https://cdn.paddle.com/paddle/paddle.js"
            document.getElementsByTagName("head")[0].appendChild(script)
          }}
        >
          {/* eslint-disable-next-line no-nested-ternary */}
          {isSubscribed ? (
            <Trans>Subscribed</Trans>
          ) : loading || user.isLoading ? (
            <LoadingIndicator color="onPrimary" />
          ) : (
            <Trans>Subscribe Now</Trans>
          )}
        </Button>
      </Card>
    </Box>
  )
}

const Feature: FC<{ text: string; comingSoon?: boolean }> = ({
  text,
  comingSoon,
}) => {
  return (
    <Flex sx={{ alignItems: "center" }}>
      <Icon as={CheckmarkIcon} fill="textPrimary" />
      <Typography.Body ml={2}>
        <Trans id={text} />
      </Typography.Body>
      {comingSoon && (
        <Typography.Body ml={2} color="textPrimary">
          <Trans>Soon</Trans>
        </Typography.Body>
      )}
    </Flex>
  )
}

export default PageSubscription
