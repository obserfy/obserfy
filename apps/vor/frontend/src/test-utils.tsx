/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
