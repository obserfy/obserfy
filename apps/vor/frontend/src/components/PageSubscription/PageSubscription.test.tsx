import { render } from "../../test-utils"
import { Basic } from "./PageSubscription.stories"

describe("PageSubscription", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
