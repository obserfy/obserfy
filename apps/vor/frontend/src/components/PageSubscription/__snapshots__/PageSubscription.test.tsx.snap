// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSubscription should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: scale(0);
    -moz-transform: scale(0);
    -ms-transform: scale(0);
    transform: scale(0);
  }

  100% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@keyframes animation-2 {
  0% {
    -webkit-transform: translate(0, 0);
    -moz-transform: translate(0, 0);
    -ms-transform: translate(0, 0);
    transform: translate(0, 0);
  }

  100% {
    -webkit-transform: translate(6px, 0);
    -moz-transform: translate(6px, 0);
    -ms-transform: translate(6px, 0);
    transform: translate(6px, 0);
  }
}

@keyframes animation-2 {
  0% {
    -webkit-transform: translate(0, 0);
    -moz-transform: translate(0, 0);
    -ms-transform: translate(0, 0);
    transform: translate(0, 0);
  }

  100% {
    -webkit-transform: translate(6px, 0);
    -moz-transform: translate(6px, 0);
    -ms-transform: translate(6px, 0);
    transform: translate(6px, 0);
  }
}

@keyframes animation-3 {
  0% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  100% {
    -webkit-transform: scale(0);
    -moz-transform: scale(0);
    -ms-transform: scale(0);
    transform: scale(0);
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 500px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-10:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-10:last-child>.splitter {
  display: none;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 16rem;
  border-radius: 16px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin: 16px;
  padding: 32px;
  margin-top: 16px;
  border-radius: 16px;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 8px;
  color: var(--theme-ui-colors-textPrimary);
  font-weight: bold;
  line-height: 1;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 48px;
  font-weight: bold;
  line-height: 1;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-left: 8px;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-top: 16px;
  margin-bottom: 16px;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 16px;
  margin-bottom: 16px;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-textPrimary);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-left: 8px;
}

.emotion-41 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 16px;
  margin-bottom: 16px;
  padding-top: 16px;
  border-top-color: var(--theme-ui-colors-border);
  border-top-width: 1px;
  border-top-style: solid;
}

.emotion-42 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-43 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
}

.emotion-44 {
  font-size: 14px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-left: auto;
}

.emotion-45 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  opacity: 0.5;
}

.emotion-46 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-47 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  font-weight: bold;
}

.emotion-48 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-left: auto;
  font-weight: bold;
  opacity: 0.5;
}

.emotion-49 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-top: 32px;
  padding: 16px;
  width: 100%;
  font-weight: bold;
  border-radius: 16px;
}

.emotion-49:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-49:hover,
.emotion-49:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-49:hover:disabled,
.emotion-49:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-50 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 4px;
  margin-right: 4px;
  display: inline-block;
  position: relative;
  width: 18px;
  height: 18px;
}

.emotion-50 div {
  position: absolute;
  top: 8px;
  width: 3px;
  height: 3px;
  border-radius: 50%;
  background-color: var(--theme-ui-colors-onPrimary);
  -webkit-animation-timing-function: cubic-bezier(0, 1, 1, 0);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}

.emotion-50 div:nth-of-type(1) {
  left: 2px;
  -webkit-animation: animation-1 0.6s infinite;
  animation: animation-1 0.6s infinite;
}

.emotion-50 div:nth-of-type(2) {
  left: 2px;
  -webkit-animation: animation-2 0.6s infinite;
  animation: animation-2 0.6s infinite;
}

.emotion-50 div:nth-of-type(3) {
  left: 8px;
  -webkit-animation: animation-2 0.6s infinite;
  animation: animation-2 0.6s infinite;
}

.emotion-50 div:nth-of-type(4) {
  left: 14px;
  -webkit-animation: animation-3 0.6s infinite;
  animation: animation-3 0.6s infinite;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/admin"
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/admin"
          >
            <span
              class="emotion-8"
            >
              Admin
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-10"
          >
            Plans & Billing
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-11"
      >
        <div
          class="emotion-12"
        />
      </div>
      <div
        class="emotion-13"
      >
        <p
          class="emotion-14"
        >
          Standard Plan
        </p>
        <div
          class="emotion-15"
        >
          <h3
            class="emotion-16"
          >
            $4
          </h3>
          <p
            class="emotion-17"
          >
            User / Month
          </p>
        </div>
        <p
          class="emotion-18"
        >
          Get full access to all features.
           
          <a
            href="https://obserfy.com/pricing"
          >
            <u>
              Learn more.
            </u>
          </a>
        </p>
        <div
          class="emotion-19"
        >
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Unlimited students
            </p>
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Record observations
            </p>
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Create lesson plans
            </p>
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Track curriculum progress
            </p>
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Parent dashboard
            </p>
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Image gallery
            </p>
          </div>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Video gallery
            </p>
          </div>
        </div>
        <div
          class="emotion-41"
        >
          <div
            class="emotion-42"
          >
            <p
              class="emotion-43"
            >
              You currently have
            </p>
            <a
              class="emotion-44"
              href="/dashboard/admin/users"
            >
              Manage users
            </a>
          </div>
          <p
            class="emotion-45"
          >
            ... 
             User(s)
          </p>
        </div>
        <div
          class="emotion-46"
        >
          <p
            class="emotion-47"
          >
            Total
          </p>
          <p
            class="emotion-48"
          >
            ... 
            $ / month
          </p>
        </div>
        <button
          class="emotion-49"
          disabled=""
        >
          <div
            class="emotion-50"
          >
            <div />
            <div />
            <div />
            <div />
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
